src/audible_cli/models.py

```diff
@@ -95,13 +95,18 @@ def create_base_filename(self, mode: str):
         elif "unicode" in mode:
             base_filename = unicodedata.normalize("NFKD", self.full_title or "")
 
+        # asin_only
         else:
             base_filename = self.asin
 
-        if "asin" in mode:
+        if "asin" in mode and not "asin_only" == mode:
             base_filename = self.asin + "_" + base_filename
 
-        return base_filename
+        encoded = base_filename.encode('utf-8')
```

Comment:
Why do you have encoded the filename to bytes before you slice?

Reply:
because most filesystems are limited to byte numbers, not to characters. UTF-8 characters might need up to 4 bytes.

Reply:
and luckily this code works in case of ascii as well, because ascii is a subset of utf-8

Reply:
and this is the reason why your suffixes (`-annotations.json`) are safe, because you know their maximum length, and you know they are ascii.

Reply:
> because most filesystems are limited to byte numbers, not to characters. UTF-8 characters might need up to 4 bytes (think chinese characters).

I does not was aware about this behavior. Thank you for this "lesson". 
On `ascii` or `asin_ascii` mode, the filename is get by `BaseItem.full_title_slugify` property. So there are only `ASCII` letters. But I will re-add the byte encoding now.

Reply:
what I meant by "safe": when you use UTF-8 and you have a chinese book name and you cut that off at 230 bytes. And then you add your suffix (`annotations.json`) and since your suffix **consists of** only ASCII character, each character will only add 1 byte. And so including your suffix the complete filename will be guaranteed to be less than 254 bytes.